pkg/kube/inject/inject.go

```diff
@@ -857,6 +857,12 @@ func IntoObject(sidecarTemplate string, valuesConfig string, meshconfig *meshcon
 	}
 
 	metadata.Annotations[annotation.SidecarStatus.Name] = status
+	if status != "" && metadata.Labels[model.MTLSReadyLabelName] == "" {
```

Comment:
nit: why check `status`?

Reply:
@hzxuzhonghu perhaps being overcautious.  If the istio status has been set to something, safe to assume the pod has been injected with everything needed to label it as mtlsReady